src/aleph/http/http2.clj

```diff
@@ -0,0 +1,503 @@
+                                            (FileChannel/open
+                                              ^Path body
+                                              (into-array OpenOption [StandardOpenOption/READ]))
+                                            ^int file-chunk-size))
+
+        (instance? HttpFile body)
+        (send-http-file ch stream headers body)
+
+        (instance? FileChannel body)
+        (send-chunked-body ch stream headers
+                           (ChunkedNioFile. ^FileChannel body ^int file-chunk-size))
+
+        (instance? FileRegion body)
+        (if (or (-> ch (.pipeline) (.get ^Class SslHandler))
+                (some-> ch (.parent) (.pipeline) (.get ^Class SslHandler)))
+          (let [emsg (str "FileRegion not supported with SSL in Netty")
+                ex (ex-info emsg {:ch ch :headers headers :body body})
+                e (Http2FrameStreamException. stream Http2Error/PROTOCOL_ERROR ex)]
+            (log/error e emsg)
+            (netty/close ch))
+          (send-file-region ch headers body))
+
+        :else
+        (try
+          (send-streaming-body ch stream headers body chunk-size)
+          (catch Throwable e
+            (log/error e "Error sending body of type " (StringUtil/simpleClassName body))
+            (throw e))))
+
+      (catch Exception e
+        (log/error e "Error sending message")
+        (throw (Http2FrameStreamException. (.stream ch)
+                                           Http2Error/PROTOCOL_ERROR
+                                           (ex-info "Error sending message" {:headers headers :body body} e)))))))
+
+;; NOTE: can't be as vague about whether we're working with a channel or context in HTTP/2 code,
+;; because we need access to the .stream method. We have a lot of code in aleph.netty that
+;; branches based on the class (channel vs context), but that's not ideal. It's slower, and
+;; writing to the channel vs the context means different things, anyway, they're not
+;; usually interchangeable.
+(defn send-request
+  [^Http2StreamChannel ch req response]
+  (log/trace "http2 send-request fired")
+
+  (when (multipart/is-multipart? req)
+    ;; shouldn't reach at the moment, but just in case
+    (throw (ex-info "HTTP/2 code path multipart not supported"
+                    {:req req
+                     :stream (.stream ch)})))
+
+  (try
+    (let [body (if (= :trace (:request-method req))
+                 ;; RFC #7231 4.3.8. TRACE
+                 ;; A client MUST NOT send a message body...
+                 (do
+                   (when (some? (:body req))
+                     (log/warn "Non-nil TRACE request body was removed"))
+                   nil)
+                 (:body req))
+          headers (ring-map->netty-http2-headers req)
+          chunk-size (or (:chunk-size req) *default-chunk-size*)
+          file-chunk-size (p/int (or (:chunk-size req) file/*default-file-chunk-size*))]
```

Comment:
Sure, if you want to set the global default. "Mutable" globals that are set once at startup time and only read from then on, don't really pose a problem. 

Otherwise you have to set a param and attach it to every map everywhere which wastes memory and clutters stuff up.